spec/email_validator_spec.rb

```diff
@@ -3,7 +3,12 @@
 describe EmailValidator do
   email_class = Class.new do
     include ActiveModel::Validations
+
     attr_accessor :email
+
+    def self.model_name
+      ActiveModel::Name.new(self, nil, "TestModel")
+    end
```

Comment:
This is required for the error message locale translating by type

Please see the `ActiveModel::Error.generate_message` method https://github.com/rails/rails/blob/main/activemodel/lib/active_model/error.rb#L80

And the documentation https://api.rubyonrails.org/classes/ActiveModel/Errors.html which says 
```Ruby
# Required dependency for ActiveModel::Errors
  extend ActiveModel::Naming
```

Since it's an anonymous class, we can't just extend ActiveModel::Naming